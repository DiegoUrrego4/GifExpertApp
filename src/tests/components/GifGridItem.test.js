import React from 'react';
import { shallow } from 'enzyme/build';
import GifGridItem from '../../components/GifGridItem';

describe('Tests in GifGridItem', () => {
  const title = 'Un t√≠tulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('The component must be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('It should have a paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Must have the image equal to the url and alt of the props.', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Must have class animate__fadeInUp', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeInUp')).toBe(true);
  });
});
