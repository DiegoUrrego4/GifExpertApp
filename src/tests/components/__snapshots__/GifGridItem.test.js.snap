// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifGridItem The component must be displayed correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
